### PR TITLE
Add support for Twitter's Finatra application framework

### PR DESCRIPTION
**Pull request by timkral**
CC: @thespags 

This is some work that I've done to allow more Scala development with `appbuilder`. Whereas we have a pretty clear Java development story with Dropwizard, there wasn't a clear story for Scala. Of course, you could use Dropwizard with Scala, but chances are that Scala people won't like that very much. Finatra, on the other hand, is a framework native to Scala so adding support there creates a more clear story.

Included here are some nice ftests which test guice injection for a Finatra app created through `appbuilder`.

### CODE REVIEW

File: app-finatra-test/src/test/scala/net/spals/appbuilder/app/finatra/plugins/PluginsFinatraWebApp.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  setServiceConfigFromClasspath("config/plugins-finatra-service.conf")
+  setServiceScan(new ServiceScan.Builder()
+    .addServicePackages("net.spals.appbuilder.app.finatra.plugins")
+    .addServicePlugins("net.spals.appbuilder.mapstore.cassandra", classOf[MapStore])
+    .addServicePlugins("net.spals.appbuilder.mapstore.dynamodb", classOf[MapStore])
+    .addServicePlugins("net.spals.appbuilder.model.protobuf", classOf[ModelSerializer])
+    .build())
+  build()
+
+}
```
Comment (timkral) [inline app-finatra-test/src/test/scala/net/spals/appbuilder/app/finatra/plugins/PluginsFinatraWebApp.scala:24]: Sample application which uses `appbuilder` plugins.

File: app-finatra-test/src/test/scala/net/spals/appbuilder/app/finatra/sample/SampleFinatraWebApp.scala [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    .addDefaultServices(classOf[FileStore])
+    .addDefaultServices(classOf[MapStore])
+    .addDefaultServices(classOf[MessageConsumer], classOf[MessageProducer])
+    .addDefaultServices(classOf[ModelSerializer])
+    .build())
+  addModule(new SampleGuiceModule)
+  addModule(new SampleTwitterModule)
+  build()
+
+}
```
Comment (timkral) [inline app-finatra-test/src/test/scala/net/spals/appbuilder/app/finatra/sample/SampleFinatraWebApp.scala:32]: Sample application which uses the `appbuilder` framework with Finatra support. This is only default services (i.e. no plugins).

File: config/src/main/java/net/spals/appbuilder/config/service/ServiceScan.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ * to the scan.
+ *
+ * @author tkral
+ */
+@FreeBuilder
+public interface ServiceScan {
+
+    Set<String> getServicePackages();
+
+    Reflections getReflections();
```
Comment (timkral) [inline config/src/main/java/net/spals/appbuilder/config/service/ServiceScan.java:24]: Wrapping `Reflections` in our own `ServiceScan` class. This provides a better development experience when building the underlying `Reflections` scanner.

**Comment by thespags** [comment]
Why do you call addDefaultServices so many times instead of once?

**Comment by timkral** [comment]
An an example to show that it can be done that way if an author so chooses.

**Pull request merged**